src/main/java/org/javarosa/core/model/data/helper/Selection.java

```diff
@@ -120,7 +120,7 @@ public String getValue () {
         if (xmlValue != null && xmlValue.length() > 0) {
             return xmlValue;
         } else {
-            throw new RuntimeException("don't know xml value! perhaps selection was stored as index only and has not yet been linked up to a formdef?");
+            throw new RuntimeException("This choice is not valid. If you are using an external file, please make sure it has the expected columns (usually name and label).");
```

Comment:
This is going to be extremely rare. The only case I can think of it happening in is https://github.com/getodk/javarosa/pull/635/commits/bc2a8f26efefda828f9774f59f6bd92384e02116#diff-99f7f5318b0381156d5214c96eb7c81a107e2b9e73565eabcc606a96c7246b23R203